Source/AliveLibAE/Scrab.cpp

```diff
@@ -1634,52 +1614,50 @@ __int16 Scrab::AI_ChasingEnemy_State_Running_2(BaseAliveGameObject* pObj)
                 field_108_next_motion = eScrabMotions::M_AttackSpin_32_4A8DC0;
                 return AI_ChasingEnemy::eState1_KilledPossessedScrab_16;
             }
+        }
+
+        if (vIsObjNearby_4253B0(ScaleToGridSize_4498B0(field_CC_sprite_scale) * FP_FromInteger(3), pObj)
+            && field_106_current_motion == eScrabMotions::M_Run_2_4A89C0
+            && vOnSameYLevel_425520(pObj))
+        {
+            if (WallHit_408750(field_CC_sprite_scale * FP_FromInteger(45), pObj->field_B8_xpos - field_B8_xpos))
+            {
+                field_108_next_motion = eScrabMotions::M_Stand_0_4A8220;
+                return AI_ChasingEnemy::eState1_Idle_1;
+            }
             else
             {
-                if (vIsObjNearby_4253B0(ScaleToGridSize_4498B0(field_CC_sprite_scale) * FP_FromInteger(3), pObj)
-                    && field_106_current_motion == eScrabMotions::M_Run_2_4A89C0
-                    && vOnSameYLevel_425520(pObj))
-                {
-                    if (WallHit_408750(field_CC_sprite_scale * FP_FromInteger(45), pObj->field_B8_xpos - field_B8_xpos))
-                    {
-                        field_108_next_motion = eScrabMotions::M_Stand_0_4A8220;
-                        return AI_ChasingEnemy::eState1_Idle_1;
-                    }
-                    else
-                    {
-                        field_108_next_motion = eScrabMotions::M_AttackLunge_37_4AA0B0;
-                        return AI_ChasingEnemy::eState1_Attacking_8;
-                    }
-                }
-                else
-                {
-                    if (field_106_current_motion != eScrabMotions::M_JumpToFall_8_4A9220)
-                    {
-                        return field_11C_sub_state;
-                    }
-                    return AI_ChasingEnemy::eState1_Falling_5;
-                }
+                field_108_next_motion = eScrabMotions::M_AttackLunge_37_4AA0B0;
+                return AI_ChasingEnemy::eState1_Attacking_8;
             }
         }
         else
         {
-            if (gMap_5C3030.Is_Point_In_Current_Camera_4810D0(field_C2_lvl_number, field_C0_path_number, field_B8_xpos, field_BC_ypos, 0))
-            {
-                field_108_next_motion = eScrabMotions::M_Turn_3_4A91A0;
-            }
-            else
+            if (field_106_current_motion != eScrabMotions::M_JumpToFall_8_4A9220)
             {
-                field_106_current_motion = eScrabMotions::M_Turn_3_4A91A0;
-                field_108_next_motion = -1;
-                MapFollowMe_408D10(TRUE);
+                //A patch workaround
```

Comment:
what does this mean?

Reply:
It's just that this scenario should probably never happen in the first place, the condition is like a plaster on it.

Reply:
I forgot to finish that comment :tapeDan:

Reply:
maybe reword it cause I had no idea what it meant ⚰️